diff --git a/src/luaconf.h b/src/luaconf.h
index f47f0680..27ddba30 100644
--- a/src/luaconf.h
+++ b/src/luaconf.h
@@ -86,7 +86,7 @@
 #define LUA_QS		LUA_QL("%s")
 
 /* Various tunables. */
-#define LUAI_MAXSTACK	65500	/* Max. # of stack slots for a thread (<64K). */
+#define LUAI_MAXSTACK	33536000	/* Max. # of stack slots for a thread (<64K). */
 #define LUAI_MAXCSTACK	8000	/* Max. # of stack slots for a C func (<10K). */
 #define LUAI_GCPAUSE	200	/* Pause GC until memory is at 200%. */
 #define LUAI_GCMUL	200	/* Run GC at 200% of allocation speed. */
